File: src/main/rendering/Color.java
package rendering;

public class Color {

  public final float red;
  public final float green;
  public final float blue;
  public final float alpha;

  private Color(float red, float green, float blue, float alpha) {
    this.red = red;
    this.green = green;
    this.blue = blue;
    this.alpha = alpha;
  }

  public static Color create(float red, float green, float blue, float alpha) {
    return new Color(red, green, blue, alpha);
  }

  public static Color create(float red, float green, float blue) {
    return new Color(red, green, blue, 1f);
  }

  public static Color of(Preset preset) {
    return create(preset.red, preset.green, preset.blue);
  }

  public enum Preset {

    RED(1, 0, 0),
    GREEN(0, 1, 0),
    BLUE(0, 0, 1),
    WHITE(1, 1, 1),
    BLACK(0, 0, 0);

    float red, green, blue;

    Preset(float red, float green, float blue) {
      this.red = red;
      this.green = green;
      this.blue = blue;
    }

  }

}

File: src/main/kotlin/rendering/Color.kt
package rendering

enum class Color(val r: Float, val g: Float, val b: Float) {

  WHITE(1f, 1f, 1f),
  VERY_LIGHT_GRAY(.925f, .925f, .925f),
  LIGHT_GRAY(.75f, .75f, .75f),
  GRAY(.5f, .5f, .5f),
  DARK_GRAY(.25f, .25f, .25f),
  VERY_DARK_GRAY(.075f, .075f, .075f),
  BLACK(0f, 0f, 0f),

  FULL_RED(1f, 0f, 0f),
  FULL_GREEN(0f, 1f, 0f),
  FULL_BLUE(0f, 0f, 1f)

}